### ISSUE TITLE
Usage of namedtuple._replace should not generate warning

### ISSUE THREAD

**Issue by matangover** [microsoft/pylance-release#1454]
## Environment data

-   Language Server version: 2021.6.1
-   OS and version: macOS 10.15.7
-   Python version: 3.8

## Expected behaviour

Pylance should not warn on when calling `_replace` on a `namedtuple`.

## Actual behaviour

Pylance generates a warning: `"_replace" is protected and used outside of the class in which it is declared`.

## Code Snippet

```python
from typing import NamedTuple
class MyTuple(NamedTuple):
    field1: int
    field2: str

MyTuple(1, 2)._replace(field1=3)
```

## Notes

The `_replace` method is [documented](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace), along with other fields that begin with underscore and are not private. I can guess that the reason why the prepended an underscore to these members' names is to not clash with user-defined field names in the namedtuple.

Probably Pylance should keep a whitelist of members that start with underscore but are not private.

**Comment by erictraut** [comment]
We generally avoid hard-coding knowledge of specific classes, functions or members within Pylance because it's not a scalable approach. We might consider doing it in very special circumstances if we receive a lot of feedback.

In this case, the error is emitted only if you enable the `reportPrivateUsage` diagnostic rule, which is off by default. So you have the option of not enabling this rule, silencing it for a file using a `# pyright: reportPrivateUsage=false` comment or silencing it on a line using a `# type: ignore` comment.

**Comment by erictraut** [comment]
Thinking about this a bit more, the `_replace` method is declared in the typeshed std stubs. Type stubs are meant to document the public interface contract for a library, and library authors will generally choose to omit private methods and attributes because they are considered "implementation details". In cases where a library author explicitly chooses to include a method or attribute whose name starts with an underscore, I think it's reasonable to assume that they intend for it to be a public symbol. Following that logic, I've changed the logic for the `reportPrivateUsage` diagnostic check to suppress the check if the method or attribute is declared within a type stub file.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.6.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202162-16-june-2021

**Comment by silriozs** [comment]
> This issue has been fixed in version 2021.6.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202162-16-june-2021

This change doen't seem to include the type name declared in stub files. Some type names, e.g., `hashlib._Hash` and `curses._CursesWindow`, are still reported `reportPrivateUsage` when used. And they are important for writing type annotations for objects come from those libraries.

Language Server version: Pylance language server 2022.1.5 (pyright 8d096292)

**Comment by erictraut** [comment]
This change was specifically for private class members. I think it's reasonable to add support for module-level symbols as well. I'd appreciate it if you would file a new enhancement request to track this.
